README.md
# ukui-themes

![Check build](https://github.com/ukui/ukui-themes/workflows/Check%20build/badge.svg?branch=master)

The theme collections of UKUI.

## Getting started

### Build

Clone this repository and change to the project directory.

```
git clone https://github.com/ukui/ukui-themes.git ukui-themes
cd ukui-themes
```

Simple run: `make`

debian/changelog
ukui-themes (3.0.1-1) unstable; urgency=medium

  * New upstream release. (Closes: #985715)

 -- handsome_feng <jianfengli@ubuntukylin.com>  Fri, 26 Mar 2021 11:21:10 +0000

ukui-themes (3.0.0-1) unstable; urgency=medium

  * New upstream release.

 -- handsome_feng <jianfengli@ubuntukylin.com>  Wed, 17 Mar 2021 14:05:27 +0000

ukui-themes (1.4.1-1) unstable; urgency=medium

  * New upstream release.
    - Define @text_view_bg. (Closes: #970982)

 -- handsome_feng <jianfengli@ubuntukylin.com>  Tue, 05 Jan 2021 03:07:22 +0000

ukui-themes (1.4.0-1) unstable; urgency=medium

  * New upstream release.

 -- handsome_feng <jianfengli@ubuntukylin.com>  Fri, 17 Jul 2020 11:42:46 +0800

ukui-themes (1.3.0-1) unstable; urgency=medium

  * New upstream release.
  * debian/control:
    - Upgrade debhelper-compat level to 13.

 -- handsome_feng <jianfengli@ubuntukylin.com>  Sun, 28 Jun 2020 17:03:12 +0800

ukui-themes (1.2.0-1) unstable; urgency=medium

  * New upstream release.
  * debian/control:
    - Bump standards-vertion to 4.5.0.
    - Set Rules-Requires-Root to no.
    - use debhelper-compat.
  * debian/rules:
    - Add export NO_PNG_PKG_MANGLE=1.

 -- handsome_feng <jianfengli@ubuntukylin.com>  Tue, 17 Mar 2020 14:33:59 +0800

ukui-themes (0+git20190530-1) unstable; urgency=medium

  * Adjust ukui-black theme.
  * Add some new icons for ukui-icon-theme.
  * Make the build reproducible. (Closes: #920595)
  * debian/control:
    - Bump standards-vertion to 4.4.0.

 -- handsome_feng <jianfengli@ubuntukylin.com>  Thu, 30 May 2019 15:04:14 +0800

ukui-themes (0+git20190105-1) unstable; urgency=medium

  * Add ukui-blue, ukui-black.
  * Remove obsolete themes.
  * Update icon theme.
  * Bump standards-version to 4.3.0.
  * Bump compat version to 12.

 -- handsome_feng <jianfengli@ubuntukylin.com>  Sat, 05 Jan 2019 14:47:03 +0800

ukui-themes (0+git20170606-2) unstable; urgency=medium

  * Use tracker.d.o team address (Closes: #899719)

 -- Aron Xu <aron@debian.org>  Mon, 09 Jul 2018 17:14:36 +0800

ukui-themes (0+git20170606-1) unstable; urgency=medium

  * Initial release. (Closes: #872970)

 -- Aron Xu <aron@debian.org>  Sun, 27 Aug 2017 14:11:59 +0800


ukui-icon-theme-default/scalable/mimetypes/test.sh
#! /bin/sh

rm -rf application-x-compress.svg
ln -s  application-archive.svg application-x-compress.svg 
rm -rf package.svg
ln -s  application-archive.svg package.svg
rm -rf application-x-compressed-tar.svg
ln -s  application-archive.svg application-x-compressed-tar.svg
rm -rf gnome-mime-application-x-compress.svg
ln -s  application-archive.svg gnome-mime-application-x-compress.svg
rm -rf gnome-package.svg
ln -s  application-archive.svg gnome-package.svg
rm -rf package-x-generic.svg
ln -s  application-archive.svg package-x-generic.svg


rm -rf zip.svg
ln -s application-archive-zip.svg zip.svg 
ln -s application-archive-zip.svg application-x-zip.svg
ln -s application-archive-zip.svg application-zip.svg 

rm -rf image-eps.svg
ln -s application-eps.svg image-eps.svg

rm -rf font.svg 
ln -s application-font.svg font.svg

rm -rf application-x-msi.svg
ln -s application-vnd.microsoft.portable-executable.svg application-x-msi.svg 

rm -rf text-x-preview.svg 
ln -s empty.svg text-x-preview.svg 

rm -rf application-vnd.android.package-archive.svg 
ln -s application-apk.svg application-vnd.android.package-archive.svg 


rm -rf text-x-java.svg
ln -s application-java.svg text-x-java.svg 
ln -s application-java.svg application-x-java-archive.svg
ln -s application-java.svg application-x-java.svg

rm -rf gnome-mime-application-x-php.package-x-generic
rm -rf text-x-php.svg
ln -s application-x-php.svg text-x-php.svg 

rm -rf application-vnd.oasis.opendocument.database-template.svg 
rm -rf application-vnd.oasis.opendocument.database.svg
ln -s libreoffice-database.svg application-vnd.oasis.opendocument.database.svg 
ln -s libreoffice-database.svg application-vnd.oasis.opendocument.database-template.svg


rm -rf application-vnd.oasis.opendocument.drawing.svg
rm -rf application-vnd.oasis.opendocument.graphics-template.svg
rm -rf application-vnd.oasis.opendocument.graphics.svg
ln -s libreoffice-drawing.svg application-vnd.oasis.opendocument.drawing.svg
ln -s libreoffice-drawing.svg application-vnd.oasis.opendocument.graphics.svg
ln -s libreoffice-drawing.svg application-vnd.oasis.opendocument.graphics-template.svg

rm -rf application-vnd.oasis.opendocument.formula-template.svg
rm -rf application-vnd.oasis.opendocument.formula.svg
ln -s libreoffice-formula.svg application-vnd.oasis.opendocument.formula-template.svg
ln -s libreoffice-formula.svg application-vnd.oasis.opendocument.formula.svg

rm -rf application-vnd.oasis.opendocument.presentation-template.svg
rm -rf application-vnd.oasis.opendocument.presentation.svg
ln -s libreoffice-presentation.svg application-vnd.oasis.opendocument.presentation-template.svg
ln -s libreoffice-presentation.svg application-vnd.oasis.opendocument.presentation.svg


rm -rf application-vnd.oasis.opendocument.spreadsheet-template.svg
rm -rf application-vnd.oasis.opendocument.spreadsheet.svg
ln -s libreoffice-spreadsheet.svg application-vnd.oasis.opendocument.spreadsheet-template.svg
ln -s libreoffice-spreadsheet.svg application-vnd.oasis.opendocument.spreadsheet.svg

rm -rf application-vnd.oasis.opendocument.text-template.svg
rm -rf application-vnd.oasis.opendocument.text.svg
ln -s libreoffice-text.svg application-vnd.oasis.opendocument.text.svg
ln -s libreoffice-text.svg application-vnd.oasis.opendocument.text-template.svg

rm -rf application-vnd.oasis.opendocument.web-template.svg
rm -rf application-vnd.oasis.opendocument.web.svg
ln -s libreoffice-web.svg application-vnd.oasis.opendocument.web-template.svg
ln -s libreoffice-web.svg application-vnd.oasis.opendocument.web.svg

rm -rf application-vnd.sun.xml.calc.svg
ln -s libreoffice-spreadsheet.svg application-vnd.sun.xml.calc.svg


rm -rf application-vnd.openxmlformats-officedocument.presentationml.presentation.svg
rm -rf application-vnd.openxmlformats-officedocument.presentationml.slideshow.svg
rm -rf application-vnd.openxmlformats-officedocument.spreadsheetml.sheet.svg
rm -rf application-vnd.openxmlformats-officedocument.wordprocessingml.document.svg

ln -s  image-eps.svg application-x-eps.svg
ln -s  application-pdf.svg application-x-gzpdf.svg

rm -rf rar.svg 
ln -s  application-rar.svg rar.svg
rm -rf application-vnd.rar.svg
ln -s  application-rar.svg application-vnd.rar.svg
ln -s  application-rar.svg appplication-x-rar.svg 
ln -s  application-rar.svg gnome-mime-application-x-rar.svg









ukui-white/gtk-3.0/css-optimizer.js
'use strict';

var fs = require('fs');
try {
	var css_data = fs.readFileSync('gtk.css', 'utf-8');
} catch (err) {
	console.log("Read css file error.");
}

var CleanCSS = require('clean-css');
var input = 'a{font-weight:bold;}'; /* for test */
var options = {
	format: {
		breaks: {
			afterAtRule: true,
			afterBlockBegins: true,
			afterBlockEnds: true,
			afterComment: true,
			afterProperty: true,
			afterRuleBegins: true,
			afterRuleEnds: true,
			beforeBlockEnds: true,
			betweenSelectors: true
		},
		indentBy: 4,
		indentWith: 'space',
		spaces: {
			aroundSelectorRelation: true,
			beforeBlockBegins: true,
			beforeValue: true
		},
		wrapAt: 80
	},
	level: {
		1: {
			all: false,
			specialComments: 'all'
		}
	}
};

var output = new CleanCSS(options).minify(css_data);
//console.log(output.styles);
fs.writeFileSync("gtk.css", output.styles);


dark-sense/cursors.sh
#!/bin/bash
  mkdir cursors
  echo "24 11 11 ../24x24/bd_double_arrow.png 0 "> cursors/bd_double_arrow.conf
  echo "24 10 15 ../24x24/bottom_left_corner.png 0 "> cursors/bottom_left_corner.conf
  echo "24 15 15 ../24x24/bottom_right_corner.png 0 "> cursors/bottom_right_corner.conf
  echo "24 13 18 ../24x24/bottom_side.png 0 "> cursors/bottom_side.conf
  echo "24 12 19 ../24x24/bottom_tee.png 0 "> cursors/bottom_tee.conf
  echo "24 4 4 ../24x24/circle.png 0 "> cursors/circle.conf
  echo "24 3 3 ../24x24/color-picker.png 0 "> cursors/color-picker.conf
  echo "24 4 4 ../24x24/copy.png 0 "> cursors/copy.conf
  echo "24 11 11 ../24x24/cross.png 0 "> cursors/cross.conf
  echo "24 12 12 ../24x24/crossed_circle.png 0 "> cursors/crossed_circle.conf
  echo "24 11 11 ../24x24/crosshair.png 0 "> cursors/crosshair.conf
  echo "24 9 9 ../24x24/dnd-ask.png 0 "> cursors/dnd-ask.conf
  echo "24 9 9 ../24x24/dnd-copy.png 0 "> cursors/dnd-copy.conf
  echo "24 9 9 ../24x24/dnd-link.png 0 "> cursors/dnd-link.conf
  echo "24 9 9 ../24x24/dnd-move.png 0 "> cursors/dnd-move.conf
  echo "24 12 11 ../24x24/dnd-none.png 0 "> cursors/dnd-none.conf
  echo "24 11 11 ../24x24/dotbox.png 0 "> cursors/dotbox.conf
  echo "24 11 11 ../24x24/fd_double_arrow.png 0 "> cursors/fd_double_arrow.conf
  echo "24 12 11 ../24x24/grabbing.png 0 "> cursors/grabbing.conf
  echo "24 8 5 ../24x24/hand2.png 0 "> cursors/hand2.conf
  echo "24 4 4 ../24x24/left_ptr.png 0 "> cursors/left_ptr.conf
  echo "24 6 13 ../24x24/left_side.png 0 "> cursors/left_side.conf
  echo "24 6 12 ../24x24/left_tee.png 0 "> cursors/left_tee.conf
  echo "24 4 4 ../24x24/link.png 0 "> cursors/link.conf
  echo "24 4 19 ../24x24/ll_angle.png 0 "> cursors/ll_angle.conf
  echo "24 20 19 ../24x24/lr_angle.png 0 "> cursors/lr_angle.conf
  echo "24 12 11 ../24x24/move.png 0 "> cursors/move.conf
  echo "24 7 21 ../24x24/pencil.png 0 "> cursors/pencil.conf
  echo "24 10 11 ../24x24/plus.png 0 "> cursors/plus.conf
  echo "24 12 21 ../24x24/question_arrow.png 0 "> cursors/question_arrow.conf
  echo "24 21 4 ../24x24/right_ptr.png 0 "> cursors/right_ptr.conf
  echo "24 19 13 ../24x24/right_side.png 0 "> cursors/right_side.conf
  echo "24 20 12 ../24x24/right_tee.png 0 "> cursors/right_tee.conf
  echo "24 12 19 ../24x24/sb_down_arrow.png 0 "> cursors/sb_down_arrow.conf
  echo "24 12 12 ../24x24/sb_h_double_arrow.png 0 "> cursors/sb_h_double_arrow.conf
  echo "24 6 12 ../24x24/sb_left_arrow.png 0 "> cursors/sb_left_arrow.conf
  echo "24 19 12 ../24x24/sb_right_arrow.png 0 "> cursors/sb_right_arrow.conf
  echo "24 12 3 ../24x24/sb_up_arrow.png 0 "> cursors/sb_up_arrow.conf
  echo "24 12 13 ../24x24/sb_v_double_arrow.png 0 "> cursors/sb_v_double_arrow.conf
  echo "24 12 12 ../24x24/tcross.png 0 "> cursors/tcross.conf
  echo "24 10 10 ../24x24/top_left_corner.png 0 "> cursors/top_left_corner.conf
  echo "24 15 10 ../24x24/top_right_corner.png 0 "> cursors/top_right_corner.conf
  echo "24 13 6 ../24x24/top_side.png 0 "> cursors/top_side.conf
  echo "24 12 5 ../24x24/top_tee.png 0 "> cursors/top_tee.conf
  echo "24 4 5 ../24x24/ul_angle.png 0 "> cursors/ul_angle.conf
  echo "24 20 5 ../24x24/ur_angle.png 0 "> cursors/ur_angle.conf
  echo "24 12 12 ../24x24/X_cursor.png 0 "> cursors/X_cursor.conf
  echo "24 11 12 ../24x24/xterm.png 0 "> cursors/xterm.conf
  echo "24 11 10 ../24x24/zoom-in.png 0 "> cursors/zoom-in.conf
  echo "24 11 10 ../24x24/zoom-out.png 0 "> cursors/zoom-out.conf

  echo "32 14 14 ../32x32/bd_double_arrow.png 0 ">> cursors/bd_double_arrow.conf
  echo "32 13 19 ../32x32/bottom_left_corner.png 0 ">> cursors/bottom_left_corner.conf
  echo "32 19 19 ../32x32/bottom_right_corner.png 0 ">> cursors/bottom_right_corner.conf
  echo "32 17 23 ../32x32/bottom_side.png 0 ">> cursors/bottom_side.conf
  echo "32 16 26 ../32x32/bottom_tee.png 0 ">> cursors/bottom_tee.conf
  echo "32 5 5 ../32x32/circle.png 0 ">> cursors/circle.conf
  echo "32 3 3 ../32x32/color-picker.png 0 ">> cursors/color-picker.conf
  echo "32 3 3 ../32x32/copy.png 0 ">> cursors/copy.conf
  echo "32 15 15 ../32x32/cross.png 0 ">> cursors/cross.conf
  echo "32 16 16 ../32x32/crossed_circle.png 0 ">> cursors/crossed_circle.conf
  echo "32 15 15 ../32x32/crosshair.png 0 ">> cursors/crosshair.conf
  echo "32 11 12 ../32x32/dnd-ask.png 0 ">> cursors/dnd-ask.conf
  echo "32 12 11 ../32x32/dnd-copy.png 0 ">> cursors/dnd-copy.conf
  echo "32 12 11 ../32x32/dnd-link.png 0 ">> cursors/dnd-link.conf
  echo "32 12 11 ../32x32/dnd-move.png 0 ">> cursors/dnd-move.conf
  echo "32 15 14 ../32x32/dnd-none.png 0 ">> cursors/dnd-none.conf
  echo "32 14 14 ../32x32/dotbox.png 0 ">> cursors/dotbox.conf
  echo "32 14 14 ../32x32/fd_double_arrow.png 0 ">> cursors/fd_double_arrow.conf
  echo "32 15 14 ../32x32/grabbing.png 0 ">> cursors/grabbing.conf
  echo "32 10 6 ../32x32/hand2.png 0 ">> cursors/hand2.conf
  echo "32 5 5 ../32x32/left_ptr.png 0 ">> cursors/left_ptr.conf
  echo "32 8 17 ../32x32/left_side.png 0 ">> cursors/left_side.conf
  echo "32 8 16 ../32x32/left_tee.png 0 ">> cursors/left_tee.conf
  echo "32 5 5 ../32x32/link.png 0 ">> cursors/link.conf
  echo "32 5 25 ../32x32/ll_angle.png 0 ">> cursors/ll_angle.conf
  echo "32 26 25 ../32x32/lr_angle.png 0 ">> cursors/lr_angle.conf
  echo "32 16 15 ../32x32/move.png 0 ">> cursors/move.conf
  echo "32 9 28 ../32x32/pencil.png 0 ">> cursors/pencil.conf
  echo "32 13 15 ../32x32/plus.png 0 ">> cursors/plus.conf
  echo "32 16 27 ../32x32/question_arrow.png 0 ">> cursors/question_arrow.conf
  echo "32 27 5 ../32x32/right_ptr.png 0 ">> cursors/right_ptr.conf
  echo "32 25 17 ../32x32/right_side.png 0 ">> cursors/right_side.conf
  echo "32 26 16 ../32x32/right_tee.png 0 ">> cursors/right_tee.conf
  echo "32 15 25 ../32x32/sb_down_arrow.png 0 ">> cursors/sb_down_arrow.conf
  echo "32 16 15 ../32x32/sb_h_double_arrow.png 0 ">> cursors/sb_h_double_arrow.conf
  echo "32 7 15 ../32x32/sb_left_arrow.png 0 ">> cursors/sb_left_arrow.conf
  echo "32 25 15 ../32x32/sb_right_arrow.png 0 ">> cursors/sb_right_arrow.conf
  echo "32 16 3 ../32x32/sb_up_arrow.png 0 ">> cursors/sb_up_arrow.conf
  echo "32 15 17 ../32x32/sb_v_double_arrow.png 0 ">> cursors/sb_v_double_arrow.conf
  echo "32 16 15 ../32x32/tcross.png 0 ">> cursors/tcross.conf
  echo "32 13 13 ../32x32/top_left_corner.png 0 ">> cursors/top_left_corner.conf
  echo "32 20 13 ../32x32/top_right_corner.png 0 ">> cursors/top_right_corner.conf
  echo "32 17 7 ../32x32/top_side.png 0 ">> cursors/top_side.conf
  echo "32 16 6 ../32x32/top_tee.png 0 ">> cursors/top_tee.conf
  echo "32 5 6 ../32x32/ul_angle.png 0 ">> cursors/ul_angle.conf
  echo "32 26 6 ../32x32/ur_angle.png 0 ">> cursors/ur_angle.conf
  echo "32 15 16 ../32x32/X_cursor.png 0 ">> cursors/X_cursor.conf
  echo "32 14 15 ../32x32/xterm.png 0 ">> cursors/xterm.conf
  echo "32 14 13 ../32x32/zoom-in.png 0 ">> cursors/zoom-in.conf
  echo "32 14 13 ../32x32/zoom-out.png 0 ">> cursors/zoom-out.conf

  echo "48 11 11 ../48x48/bd_double_arrow.png 0 ">> cursors/bd_double_arrow.conf
  echo "48 19 29 ../48x48/bottom_left_corner.png 0 ">> cursors/bottom_left_corner.conf
  echo "48 29 29 ../48x48/bottom_right_corner.png 0 ">> cursors/bottom_right_corner.conf
  echo "48 25 35 ../48x48/bottom_side.png 0 ">> cursors/bottom_side.conf
  echo "48 12 19 ../48x48/bottom_tee.png 0 ">> cursors/bottom_tee.conf
  echo "48 7 7 ../48x48/circle.png 0 ">> cursors/circle.conf
  echo "48 4 4 ../48x48/color-picker.png 0 ">> cursors/color-picker.conf
  echo "48 7 7 ../48x48/copy.png 0 ">> cursors/copy.conf
  echo "48 22 22 ../48x48/cross.png 0 ">> cursors/cross.conf
  echo "48 24 24 ../48x48/crossed_circle.png 0 ">> cursors/crossed_circle.conf
  echo "48 22 22 ../48x48/crosshair.png 0 ">> cursors/crosshair.conf
  echo "48 17 17 ../48x48/dnd-ask.png 0 ">> cursors/dnd-ask.conf
  echo "48 17 17 ../48x48/dnd-copy.png 0 ">> cursors/dnd-copy.conf
  echo "48 17 17 ../48x48/dnd-link.png 0 ">> cursors/dnd-link.conf
  echo "48 17 17 ../48x48/dnd-move.png 0 ">> cursors/dnd-move.conf
  echo "48 23 21 ../48x48/dnd-none.png 0 ">> cursors/dnd-none.conf
  echo "48 21 21 ../48x48/dotbox.png 0 ">> cursors/dotbox.conf
  echo "48 21 21 ../48x48/fd_double_arrow.png 0 ">> cursors/fd_double_arrow.conf
  echo "48 23 21 ../48x48/grabbing.png 0 ">> cursors/grabbing.conf
  echo "48 15 9 ../48x48/hand2.png 0 ">> cursors/hand2.conf
  echo "48 7 7 ../48x48/left_ptr.png 0 ">> cursors/left_ptr.conf
  echo "48 11 25 ../48x48/left_side.png 0 ">> cursors/left_side.conf
  echo "48 11 24 ../48x48/left_tee.png 0 ">> cursors/left_tee.conf
  echo "48 7 7 ../48x48/link.png 0 ">> cursors/link.conf
  echo "48 7 37 ../48x48/ll_angle.png 0 ">> cursors/ll_angle.conf
  echo "48 39 37 ../48x48/lr_angle.png 0 ">> cursors/lr_angle.conf
  echo "48 24 22 ../48x48/move.png 0 ">> cursors/move.conf
  echo "48 13 41 ../48x48/pencil.png 0 ">> cursors/pencil.conf
  echo "48 20 22 ../48x48/plus.png 0 ">> cursors/plus.conf
  echo "48 23 41 ../48x48/question_arrow.png 0 ">> cursors/question_arrow.conf
  echo "48 41 7 ../48x48/right_ptr.png 0 ">> cursors/right_ptr.conf
  echo "48 37 25 ../48x48/right_side.png 0 ">> cursors/right_side.conf
  echo "48 39 24 ../48x48/right_tee.png 0 ">> cursors/right_tee.conf
  echo "48 23 37 ../48x48/sb_down_arrow.png 0 ">> cursors/sb_down_arrow.conf
  echo "48 24 23 ../48x48/sb_h_double_arrow.png 0 ">> cursors/sb_h_double_arrow.conf
  echo "48 11 23 ../48x48/sb_left_arrow.png 0 ">> cursors/sb_left_arrow.conf
  echo "48 37 23 ../48x48/sb_right_arrow.png 0 ">> cursors/sb_right_arrow.conf
  echo "48 23 5 ../48x48/sb_up_arrow.png 0 ">> cursors/sb_up_arrow.conf
  echo "48 23 25 ../48x48/sb_v_double_arrow.png 0 ">> cursors/sb_v_double_arrow.conf
  echo "48 23 23 ../48x48/tcross.png 0 ">> cursors/tcross.conf
  echo "48 19 19 ../48x48/top_left_corner.png 0 ">> cursors/top_left_corner.conf
  echo "48 29 19 ../48x48/top_right_corner.png 0 ">> cursors/top_right_corner.conf
  echo "48 25 11 ../48x48/top_side.png 0 ">> cursors/top_side.conf
  echo "48 24 9 ../48x48/top_tee.png 0 ">> cursors/top_tee.conf
  echo "48 7 9 ../48x48/ul_angle.png 0 ">> cursors/ul_angle.conf
  echo "48 39 9 ../48x48/ur_angle.png 0 ">> cursors/ur_angle.conf
  echo "48 23 23 ../48x48/X_cursor.png 0 ">> cursors/X_cursor.conf
  echo "48 21 23 ../48x48/xterm.png 0 ">> cursors/xterm.conf
  echo "48 22 19 ../48x48/zoom-in.png 0 ">> cursors/zoom-in.conf
  echo "48 22 19 ../48x48/zoom-out.png 0 ">> cursors/zoom-out.conf

  echo "24 0 0 ../24x24/left_ptr_watch_1.png 50" > cursors/left_ptr_watch.conf
  echo "24 0 0 ../24x24/left_ptr_watch_2.png 50" > cursors/left_ptr_watch.conf
  echo "24 0 0 ../24x24/left_ptr_watch_3.png 50" > cursors/left_ptr_watch.conf
  echo "24 0 0 ../24x24/left_ptr_watch_4.png 50" > cursors/left_ptr_watch.conf
  echo "24 0 0 ../24x24/left_ptr_watch_5.png 50" > cursors/left_ptr_watch.conf

  echo "32 0 0 ../32x32/left_ptr_watch_1.png 50" >> cursors/left_ptr_watch.conf
  echo "32 0 0 ../32x32/left_ptr_watch_2.png 50" >> cursors/left_ptr_watch.conf
  echo "32 0 0 ../32x32/left_ptr_watch_3.png 50" >> cursors/left_ptr_watch.conf
  echo "32 0 0 ../32x32/left_ptr_watch_4.png 50" >> cursors/left_ptr_watch.conf
  echo "32 0 0 ../32x32/left_ptr_watch_5.png 50" >> cursors/left_ptr_watch.conf

  echo "48 0 0 ../48x48/left_ptr_watch_1.png 50" >> cursors/left_ptr_watch.conf
  echo "48 0 0 ../48x48/left_ptr_watch_2.png 50" >> cursors/left_ptr_watch.conf
  echo "48 0 0 ../48x48/left_ptr_watch_3.png 50" >> cursors/left_ptr_watch.conf
  echo "48 0 0 ../48x48/left_ptr_watch_4.png 50" >> cursors/left_ptr_watch.conf
  echo "48 0 0 ../48x48/left_ptr_watch_5.png 50" >> cursors/left_ptr_watch.conf


  echo "24 11 11 ../24x24/watch_1.png 50" > cursors/watch.conf
  echo "24 11 11 ../24x24/watch_2.png 50" > cursors/watch.conf
  echo "24 11 11 ../24x24/watch_3.png 50" > cursors/watch.conf
  echo "24 11 11 ../24x24/watch_4.png 50" > cursors/watch.conf
  echo "24 11 11 ../24x24/watch_5.png 50" > cursors/watch.conf
  echo "24 11 11 ../24x24/watch_6.png 50" > cursors/watch.conf
  echo "24 11 11 ../24x24/watch_7.png 50" > cursors/watch.conf
  echo "24 11 11 ../24x24/watch_8.png 50" > cursors/watch.conf
  echo "24 11 11 ../24x24/watch_9.png 50" > cursors/watch.conf
  echo "24 11 11 ../24x24/watch_10.png 50" > cursors/watch.conf

  echo "32 15 14 ../32x32/watch_1.png 50" >> cursors/watch.conf
  echo "32 15 14 ../32x32/watch_2.png 50" >> cursors/watch.conf
  echo "32 15 14 ../32x32/watch_3.png 50" >> cursors/watch.conf
  echo "32 15 14 ../32x32/watch_4.png 50" >> cursors/watch.conf
  echo "32 15 14 ../32x32/watch_5.png 50" >> cursors/watch.conf
  echo "32 15 14 ../32x32/watch_6.png 50" >> cursors/watch.conf
  echo "32 15 14 ../32x32/watch_7.png 50" >> cursors/watch.conf
  echo "32 15 14 ../32x32/watch_8.png 50" >> cursors/watch.conf
  echo "32 15 14 ../32x32/watch_9.png 50" >> cursors/watch.conf
  echo "32 15 14 ../32x32/watch_10.png 50" >> cursors/watch.conf

  echo "48 22 22 ../48x48/watch_1.png 50" >> cursors/watch.conf
  echo "48 22 22 ../48x48/watch_2.png 50" >> cursors/watch.conf
  echo "48 22 22 ../48x48/watch_3.png 50" >> cursors/watch.conf
  echo "48 22 22 ../48x48/watch_4.png 50" >> cursors/watch.conf
  echo "48 22 22 ../48x48/watch_5.png 50" >> cursors/watch.conf
  echo "48 22 22 ../48x48/watch_6.png 50" >> cursors/watch.conf
  echo "48 22 22 ../48x48/watch_7.png 50" >> cursors/watch.conf
  echo "48 22 22 ../48x48/watch_8.png 50" >> cursors/watch.conf
  echo "48 22 22 ../48x48/watch_9.png 50" >> cursors/watch.conf
  echo "48 22 22 ../48x48/watch_10.png 50" >> cursors/watch.conf

  cd cursors/
  xcursorgen bd_double_arrow.conf bd_double_arrow
  xcursorgen bottom_left_corner.conf bottom_left_corner
  xcursorgen bottom_right_corner.conf bottom_right_corner
  xcursorgen bottom_side.conf bottom_side
  xcursorgen bottom_tee.conf bottom_tee
  xcursorgen circle.conf circle
  xcursorgen color-picker.conf color-picker
  xcursorgen copy.conf copy
  xcursorgen cross.conf cross
  xcursorgen crossed_circle.conf crossed_circle
  xcursorgen crosshair.conf crosshair
  xcursorgen dnd-ask.conf dnd-ask
  xcursorgen dnd-copy.conf dnd-copy
  xcursorgen dnd-link.conf dnd-link
  xcursorgen dnd-move.conf dnd-move
  xcursorgen dnd-none.conf dnd-none
  xcursorgen dotbox.conf dotbox
  xcursorgen fd_double_arrow.conf fd_double_arrow
  xcursorgen grabbing.conf grabbing
  xcursorgen hand2.conf hand2
  xcursorgen left_ptr_watch.conf left_ptr_watch
  xcursorgen left_ptr.conf left_ptr
  xcursorgen left_side.conf left_side
  xcursorgen left_tee.conf left_tee
  xcursorgen link.conf link
  xcursorgen ll_angle.conf ll_angle
  xcursorgen lr_angle.conf lr_angle
  xcursorgen move.conf move
  xcursorgen pencil.conf pencil
  xcursorgen plus.conf plus
  xcursorgen question_arrow.conf question_arrow
  xcursorgen right_ptr.conf right_ptr
  xcursorgen right_side.conf right_side
  xcursorgen right_tee.conf right_tee
  xcursorgen sb_down_arrow.conf sb_down_arrow
  xcursorgen sb_h_double_arrow.conf sb_h_double_arrow
  xcursorgen sb_left_arrow.conf sb_left_arrow
  xcursorgen sb_right_arrow.conf sb_right_arrow
  xcursorgen sb_up_arrow.conf sb_up_arrow
  xcursorgen sb_v_double_arrow.conf sb_v_double_arrow
  xcursorgen tcross.conf tcross
  xcursorgen top_left_corner.conf top_left_corner
  xcursorgen top_right_corner.conf top_right_corner
  xcursorgen top_side.conf top_side
  xcursorgen top_tee.conf top_tee
  xcursorgen ul_angle.conf ul_angle
  xcursorgen ur_angle.conf ur_angle
  xcursorgen watch.conf watch
  xcursorgen X_cursor.conf X_cursor
  xcursorgen xterm.conf xterm
  xcursorgen zoom-in.conf zoom-in
  xcursorgen zoom-out.conf zoom-out
#存在软链接的鼠标图形
  ln -s hand2 e29285e634086352946a0e7090d73106
  ln -s fd_double_arrow fcf1c3c7cd4491d801f1e1c78f100000
  ln -s question_arrow d9ce0ab605698f320427677b458ad60b
  ln -s bd_double_arrow c7088f0f3e6c8088236ef8e1e3e70000
  ln -s hand2 9d800788f1b08800ae810202380a0822
  ln -s move 9081237383d90e509aa00f00170e968f
  ln -s link 640fb0e74195791501fd1ed57b41487f
  ln -s copy 6407b0e94181790501fd1e167b474872
  ln -s question_arrow 5c6cd98b3f3ebcb1f9c7f1c204630408
  ln -s move 4498f0e0c1937ffe01fd06f973665830
  ln -s left_ptr_watch 3ecb610c1bf2410f44200f48c40d3599
  ln -s link 3085a0e285430894940527032f8b26df
  ln -s sb_v_double_arrow 2870a09082c103050810ffdffffe0204
  ln -s sb_h_double_arrow 14fef782d02440884392942c11205230
  ln -s copy 1081e37283d90000800003c07f3ef6bf
  ln -s left_ptr_watch 08e8e1c95fe2fc01f976f1e063a24ccd
  ln -s crossed_circle 03b6e0fcb3499374a867c041f52298f0
  ln -s h_double_arrow 028006030e0e7ebffc7f7070c0600140
  ln -s v_double_arrow 00008160000006810000408080010102
  ln -s left_ptr arrow
  ln -s cross cross_reverse
  ln -s cross diamond_cross
  ln -s dotbox dot_box_mask
  ln -s sb_v_double_arrow double_arrow
  ln -s right_ptr draft_large
  ln -s right_ptr draft_small
  ln -s dotbox draped_box
  ln -s grabbing fleur
  ln -s hand2 hand
  ln -s hand2 hand1
  ln -s sb_h_double_arrow h_double_arrow
  ln -s question_arrow help
  ln -s dotbox icon
  ln -s question_arrow left_ptr_help
  ln -s X_cursor pirate
  ln -s dotbox target
  ln -s left_ptr top_left_arrow
  ln -s sb_v_double_arrow v_double_arrow

#
  rm *conf
  cd ..
